### PR TITLE
docs: Module 3 Queries, Events - code sample integration (B9lab updates 2-5-a)

### PR DESCRIPTION
**Pull request by coldice**
Docs content update for Module 3, Section _Queries, Events_ - part of the [b9lab-content](https://github.com/cosmos/sdk-tutorials/tree/master/b9lab-content).

This PR adds code samples to the content. The code shown here is taken from the [code example repo](https://github.com/cosmos/b9-checkers-academy-draft/) and has been hardly adjusted. The repo has been reviewed so no need to review it here. What we are looking for is to make sure that the introductory texts are not misleading or even downright wrong.

### Content state

**✔️ Completed**

### Technical review
- [x] requested
- [x] completed

### Language review
- [x] requested
- [x] completed

### CODE REVIEW

File: b9lab-content/3-main-concepts/13-events.md [hunk 1]
```diff
@@ -61,35 +63,59 @@ func NewHandler(keeper Keeper) sdk.Handler {
...
 
-* Per transaction:
-    * The game was created. Both players need to be notified.
-    * The player has played. The other player needs to be notified.
-    * The game was rejected. The other player needs to be notified.
-    * Perhaps the game was won. Both players need to be notified.
-* Per block:
-    * These games timed out. The game ids and the involved players need to be notified.
+You can filter for event types and attribute values. For example, a transfer transaction triggers an event of type `Transfer` and has `Recipient` and `Sender` as attributes (as defined in the `events.go` file of the bank module.
+
```
Comment (nooomski) [inline b9lab-content/3-main-concepts/13-events.md:86]: Not a priority, but a wrap up and CTA would be desirable.

File: b9lab-content/3-main-concepts/13-events.md [hunk 2]
```diff
@@ -61,35 +63,59 @@ func NewHandler(keeper Keeper) sdk.Handler {
...
+        sdk.NewAttribute("Red", msg.Red),
+        sdk.NewAttribute("Black", msg.Black),
+        sdk.NewAttribute("Wager", strconv.FormatUint(msg.Wager, 10)),
+        sdk.NewAttribute("Token", msg.Token),
+    ),
+)
+```
+From this model, it is easy to add events to the other transaction types, where you keep in mind that the events are meant to inform and notify relevant parties.
+
+You ought to also emit an event for games that have timed out. After all, this is part of their lifecycle. You would do it in the end blocker:
```
Comment (nooomski) [inline b9lab-content/3-main-concepts/13-events.md:108]:
```suggestion
You should also emit an event for games that have timed out. After all, this is part of their lifecycle. You would do that in the end blocker:
```

**Review by nooomski — COMMENTED** [review]
Good stuff. Feedback not mission critical, but would be great to get some fixes here next week if possible @xavierlepretre

File: b9lab-content/3-main-concepts/13-events.md [hunk 3]
```diff
@@ -22,37 +21,40 @@ message Event {
...
+* `EndBlock`
+* `CheckTx`
+* `DeliverTx`
 
 Events are managed by an abstraction called the `EventManager`.
 
 ## EventManager
 
-Eventmanager tracks a list of Events for the entire execution flow of a transaction or BeginBlock/EndBlock. EventManager implements a simple wrapper around a slice of Event objects that can be emitted from and provides useful methods. The most used method for Cosmos SDK module and application developers is `EmitEvent`.
+`EventManager` tracks a list of events for the entire execution flow of a transaction or `BeginBlock`/`EndBlock`. `EventManager` implements a simple wrapper around a slice of `Event` objects that can be emitted from and provides useful methods. The most used method for Cosmos SDK module and application developers is `EmitEvent`.
```
[the remark this answers is not in the document]
Comment (xavierlepretre) [reply b9lab-content/3-main-concepts/13-events.md:47]: I am going with:

> For each of the 4 ABCI calls, `CheckTx`, `DeliverTx`, `BeginBlock` and `EndBlock`, the `EventManager` keeps a list of events and delivers it in full as part of the ABCI response. After delivery in an ABCI response, the `EventManager` empties its list of events.

File: b9lab-content/3-main-concepts/13-events.md (hunk 1 above)
Comment (xavierlepretre) [reply b9lab-content/3-main-concepts/13-events.md:95]: Updated.

File: b9lab-content/3-main-concepts/13-events.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (nooomski) [reply b9lab-content/3-main-concepts/13-events.md]: Looks good to me, thanks!

**Pull request merged**